include/mxnet/lib_api.h

```diff
@@ -378,6 +436,13 @@ class OpResource {
     : cpu_malloc(cpu_malloc_fp), gpu_malloc(gpu_malloc_fp),
       cpu_alloc(cpu_alloc_fp), gpu_alloc(gpu_alloc_fp), cuda_stream(stream) {}
 
+  OpResource(xpu_malloc_t cpu_malloc_fp, void* cpu_alloc_fp,
+             xpu_malloc_t gpu_malloc_fp, void* gpu_alloc_fp, void* stream,
+             ndarray_malloc_t ndarray_malloc_fp, void* ndarray_alloc_fp)
```

Comment:
Do we need to have 2 constructors for this? or will we always just provide all arguments in  callFcomp functions?

Reply:
Removed the first one.